pkg/app/AppService.go

```diff
@@ -203,6 +203,12 @@ func (impl AppServiceImpl) AppendCurrentApplicationStatus(app v1alpha1.Applicati
 		if err != nil {
 			return isHealthy, err
 		}
+
+		if pipelineOverride.Pipeline.AppId != dbApp.Id {
```

Comment:
will this be enough to debug and also are we returning right  parameters? (isHealthy, nil)